pkg/cbox/utils/conversions.go

```diff
@@ -128,20 +128,8 @@ func SharePermToInt(p *provider.ResourcePermissions) int {
 
 // IntTosharePerm retrieves read/write permissions from an integer
 func IntTosharePerm(p int, itemType string) *provider.ResourcePermissions {
-	switch p {
-	case 1:
-		return conversions.NewViewerRole().CS3ResourcePermissions()
-	case 15:
-		if itemType == "folder" {
-			return conversions.NewEditorRole().CS3ResourcePermissions()
-		}
-		return conversions.NewFileEditorRole().CS3ResourcePermissions()
-	case 4:
-		return conversions.NewUploaderRole().CS3ResourcePermissions()
-	default:
-		// TODO we may have other options, for now this is a denial
-		return &provider.ResourcePermissions{}
-	}
+	perms, _ := conversions.NewPermissions(p)
```

Comment:
I do not understand the refactoring here. Can you explain?

Reply:
It was converting an `int` to a `*provider.ResourcePermissions`. But it was using a custom implementation. This is bad because it will break when somethin changes.
Other service do what we do now. Use `NewPermission` to create a new permissions struct, then use `CS3ResourcePermissions()` to convert it to the `*provider.ResourcePermissions`.
I can implement a custom logic, but I think going with the default is less error prone.

Reply:
ok, let us see what happens.